### PR TITLE
Add `indicatorColor` & `indicatorShape` to `NavigationRail`, `NavigationDrawer` and move these properties from destination to `NavigationBar`

### LINKED ISSUE #117047: Add `indicatorColor` & `indicatorShape` properties to `NavigationRail` and `NavigationDrawer`
Currently, the only way to customize `indicatorColor` & `indicatorShape` is from themes.
Add these properties to the widget themselves.

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/117047

### Description
1. Add `indicatorColor` & `indicatorShape` to `NavigationRail` & `NavigationDrawer` 
2. Move `indicatorColor` & `indicatorShape` (which were added in https://github.com/flutter/flutter/pull/116992) from `NavigationDestionation` to `NavigationBar` to the widget for consistent API since `NavigationRail` already has `indicatorColor` in the widget. 
3. Rename `_indicator` to `_getIndicatorDecoration` in test classes.
4. Add missing whitespaces and remove the extra line from test classes.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/navigation_bar.dart [hunk 1]
```diff
@@ -300,12 +308,6 @@ class NavigationDestination extends StatelessWidget {
   /// would use a size of 24.0 and [ColorScheme.onSurface].
   final Widget? selectedIcon;
 
-  /// The color of the [indicatorShape] when this destination is selected.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_bar.dart:303]: AFAICT these properties only landed yesterday - https://github.com/flutter/flutter/pull/116992

So it should be OK to remove them here.

Comment (TahaTesser) [reply packages/flutter/lib/src/material/navigation_bar.dart:303]: Yes, it was merged around 15 hours ago., that's the reason I removed them here and moved up the widget itself (`NavigationBar`) to match the `NavigationRail` API and expanded the PR to cover `NavigationDrawer` for one consistent update.

Comment (TahaTesser) [reply packages/flutter/lib/src/material/navigation_bar.dart:303]: Please let me know if there is any action needed.

File: packages/flutter/lib/src/material/navigation_bar.dart [hunk 2]
```diff
@@ -158,6 +160,12 @@ class NavigationBar extends StatelessWidget {
   /// overlay is applied.
   final Color? surfaceTintColor;
 
+  /// The color of the [indicatorShape] when this destination is selected.
+  final Color? indicatorColor;
+
+  /// The shape of the selected inidicator.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_bar.dart:166]:
```suggestion
  /// The shape of the selection indicator.
```

File: packages/flutter/lib/src/material/navigation_bar.dart [hunk 3]
```diff
@@ -588,6 +593,16 @@ class _NavigationDestinationInfo extends InheritedWidget {
   /// label, or hide all labels.
   final NavigationDestinationLabelBehavior labelBehavior;
 
+  /// The color of the indicator.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_bar.dart:596]:
```suggestion
  /// The color of the selection indicator.
```

File: packages/flutter/lib/src/material/navigation_rail.dart [hunk 4]
```diff
@@ -933,6 +940,12 @@ class NavigationRailDestination {
   ///    icons.
   final Widget selectedIcon;
 
+  /// The color of the [indicatorShape] when this destination is selected.
+  final Color? indicatorColor;
+
+  /// The shape of the selected inidicator.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_rail.dart:946]:
```suggestion
  /// The shape of the selection indicator.
```

File: packages/flutter/lib/src/material/navigation_rail.dart [hunk 5]
```diff
@@ -933,6 +940,12 @@ class NavigationRailDestination {
   ///    icons.
   final Widget selectedIcon;
 
+  /// The color of the [indicatorShape] when this destination is selected.
+  final Color? indicatorColor;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_rail.dart:944]: The indicatorColor and indicatorShape API doc should explain how they're related to NavigationRail's properties and likewise for NavigationRailTheme.

File: packages/flutter/lib/src/material/navigation_rail.dart [hunk 6]
```diff
@@ -308,6 +309,10 @@ class NavigationRail extends StatefulWidget {
   /// when [useIndicator] is true.
   final Color? indicatorColor;
 
+  /// Overrides the default value of [NavigationRail]'s selection indicator shape,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_rail.dart:312]: The indicatorColor and indicatorShape properties should explain how they're related to NavigationRail and NavigationRailTheme and what the default values are.

File: packages/flutter/lib/src/material/navigation_drawer.dart [hunk 7]
```diff
@@ -478,6 +490,16 @@ class _NavigationDrawerDestinationInfo extends InheritedWidget {
   /// to 1 (selected).
   final Animation<double> selectedAnimation;
 
+  /// The color of the indicator.
+  ///
+  /// This is used by destinations to override the indicator color.
+  final Color? indicatorColor;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/navigation_drawer.dart:496]: The indicatorColor and indicatorShape properties should explain how they're related to NavigationDrawerTheme and what the default values are.

**Comment by TahaTesser** [comment]
@HansMuller 
Updated the PR :)

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**